Code/Framework/AzCore/AzCore/Settings/ConfigParser.cpp

```diff
@@ -196,6 +196,11 @@ namespace AZ::Settings
                 memmove(configBuffer.begin(), frontIter, readOffset);
             }
 
+            if (remainingFileLength == 0)
+            {
+                // There are no more bytes to read, no need to attempt to read from the stream
+                break;
+            }
```

Comment:
Can you add this change to the [TextParser.cpp](https://github.com/o3de/o3de/blob/eebfb5e68e39a2b073f75c529aad71d2703d5fb2/Code/Framework/AzCore/AzCore/Settings/ConfigParser.cpp) as well.